lib/api/core/plugins/pluginsContext.js

```diff
@@ -1,52 +1,61 @@
 var
-  walk = require('walk'),
-  path = require('path');
-
-module.exports = function PluginContext(kuzzle) {
-  var
-    context = {
-      RequestObject: require('kuzzle-common-objects').Models.requestObject,
-      ResponseObject: require('kuzzle-common-objects').Models.responseObject,
-      NotificationObject: require('../models/notificationObject')
-    },
-    options = {
-      listeners: {
-        file: function (root, fileStats, next) {
-          var prototype = fileStats.name.substr(0, fileStats.name.length - 3);
-          if (fileStats.name.substr(-3) === '.js') {
-            prototype = prototype[0].toUpperCase() + prototype.slice(1);
-            context[prototype] = require(path.join(__dirname, '..', 'errors', fileStats.name));
-          }
-          next();
-        }
-      }
-    };
-  walk.walkSync(path.join(__dirname, '..', 'errors'), options);
-
-  // Add lazy-loading repositories getter:
-  context.repositories = function () {
-    return kuzzle.repositories;
+  util = require('util'),
+  errors = require('kuzzle-common-objects').Errors,
+  RequestObject = require('kuzzle-common-objects').Models.requestObject,
+  ResponseObject = require('kuzzle-common-objects').Models.responseObject,
+  NotificationObject = require('../models/notificationObject'),
+  _kuzzle;
+
+
+function PluginContext () {
+  this.RequestObject = RequestObject;
+  this.RequestObject = ResponseObject;
+  this.NotificationObject = NotificationObject;
+  this.Dsl = require('../../dsl');
+  this.httpPort = _kuzzle.config.httpPort;
+
+  Object.keys(errors).forEach(errorConstructor => {
+    this[errorConstructor] = errors[errorConstructor];
```

Comment:
For compatibility reasons, `this[errorConstructor]` must have its first letter capitalized: `this[_.upperFirst(errorConstructor)]` = ....`